apps/meteor/client/views/room/contextualBar/RoomMembers/AddUsers/ValidateMatrixInvitedUsers/useAddMatrixUsers.tsx

```diff
@@ -0,0 +1,66 @@
+import type { IRoom } from '@rocket.chat/core-typings';
+import { useMutableCallback } from '@rocket.chat/fuselage-hooks';
+import { useSetModal, useMethod, useToastMessageDispatch, useTranslation, useEndpoint } from '@rocket.chat/ui-contexts';
+import React from 'react';
+
+import ValidateMatrixIdModal from './ValidateMatrixIdModal';
+
+export type useAddMatrixUsersProps = {
+	rid: IRoom['_id'];
+	reload: () => void;
+	onClickBack: () => void;
+	handleUsers: (value: string | string[]) => void;
+	users: string[];
+};
+
+export const useAddMatrixUsers = ({ rid, reload, onClickBack, handleUsers, users }: useAddMatrixUsersProps) => {
+	const t = useTranslation();
+	const setModal = useSetModal();
+	const saveAction = useMethod('addUsersToRoom');
+	const dispatchToastMessage = useToastMessageDispatch();
+	const handleClose = useMutableCallback(() => setModal(null));
+	const dispatchVerifyEndpoint = useEndpoint('GET', '/v1/federation/verifyMatrixId', undefined);
+
+	const handleSave = useMutableCallback(async () => {
+		try {
+			await saveAction({ rid, users });
+			dispatchToastMessage({ type: 'success', message: t('Users_added') });
+			handleClose();
+			onClickBack();
+			reload();
+		} catch (error) {
+			dispatchToastMessage({ type: 'error', message: error as Error });
+			handleClose();
+		}
+	});
+
+	const handleMatrixValidation = useMutableCallback(async () => {
+		try {
+			const matrixIdVerificationMap = new Map();
+			const matrixIds = users.filter((user) => user.startsWith('@'));
+
+			const matrixIdsVerificationPromises = matrixIds.map((matrixId) => dispatchVerifyEndpoint({ matrixId }));
+
+			const matrixIdsVerificationPromiseResponse = await Promise.allSettled(matrixIdsVerificationPromises);
+			const matrixIdsVerificationFulfilledResults = matrixIdsVerificationPromiseResponse.filter(
+				(res) => res.status === 'fulfilled',
+			) as PromiseFulfilledResult<{ result: string }>[];
+
+			for (let i = 0; i < matrixIds.length; i++) {
```

Comment:
Maybe a map is better here? You can get the current index of a map from the parameters of the callback

Reply:
that would not work, since the purpose of the loop is to fill the HashMap `matrixIdVerificationMap` and the `.map` will expect a return value after the call.

Basically I have two arrays which I wan to store into the HashMap, one contains the keys and the other the values; they share the index position and that is what I'm using to setup the HashMap

Reply:
Then you can go with something like:

```
matrixIds.forEach((id, idx) => {
	const { value: { result } } = matrixIdsVerificationFulfilledResults[idx];
	matrixIdVerificationMap.set(id, result);
});
```

wouldn't that work? 👀

Reply:
nice!